### DISCUSSION TITLE
FBA Inventory API, getInventorySummaries - random throttling

### DISCUSSION THREAD

**Discussion by dave-sewell** [abuzuhri/Amazon-SP-API-CSharp#281] [Q&A] [answered]
I have started using the getInventorySummaries function of the FBA Inventory API, and it seems to work great except that it gets throttled in an apparently random fashion that is not related to how many records are returned. Here is the code;

            ParameterGetInventorySummaries filter = new ParameterGetInventorySummaries();
            filter.details = true;
            filter.granularityType = Granularity.GranularityTypeEnum.Marketplace;
            filter.granularityId = marketplaceId;
            filter.marketplaceIds = new List<string>();
            filter.marketplaceIds.Add(marketplaceId);
            if (status.LastDateUpdated.HasValue)
            {
                filter.startDateTime = status.LastDateUpdated.Value.UtcDateTime;
            }
            List<InventorySummaries> inventoryList = amazonConnection.FbaInventory.GetInventorySummaries(filter);
            
When the above code is called without using startDateTime, it returns about 2,400 records. Most of the time it works, but sometimes it gets rate throttled. I am making just the one call above to the library, so there must be multiple calls going on behind the scenes.

Interestingly, when I pass in a startDateTime value equal to the latest date updated (stored from previous calls), I only typically get a few records back (usually less than 5), as expected, but the rate throttling seems to occur randomly with about the same frequency as when I don't use startDateTime and get 2,400 records.

Any ideas on how to overcome the rate throttling? I suspect a bug in the library.

Thanks.

**Comment by abuzuhri** [comment, marked as answer]
I do test from my side and i cant see error , please double check if you use only one instance from the Application and not share the key with different application

**Reply by dave-sewell** [reply]
Thank you for responding so quickly. There are two instances - a production instance and a test instance. The test instance, however, was the only one making calls to FBA Inventory API. The test instance was making the calls at the rate of one per minute. However, when I disabled the test instance and moved the same code to the production instance, calling the function at the rate of once every two minutes, I do not see a throttling issue. Not sure what the difference is because once per minute is a lot slower than the published throttling rate. However, it works fine in production so this will not be a problem. Thanks again.
